core/src/main/java/org/infinispan/stream/impl/CacheIntermediatePublisher.java

```diff
@@ -26,9 +25,9 @@
  * @param <R>
  */
 public final class CacheIntermediatePublisher<R> implements ModifiedValueFunction<Publisher<Object>, Publisher<R>>, InjectableComponent {
-   private final Queue<IntermediateOperation> intOps;
+   private final Iterable<IntermediateOperation<?, ?, ?, ?>> intOps;
 
-   public CacheIntermediatePublisher(Queue<IntermediateOperation> intOps) {
+   public CacheIntermediatePublisher(Iterable<IntermediateOperation<?, ?, ?, ?>> intOps) {
```

Comment:
Please also change the other methods below

E.g.

```java
      Flowable<?> innerPublisher = Flowable.fromPublisher(objectPublisher);
      for (IntermediateOperation<?, ?, ?, ?> intOp : intOps) {
         innerPublisher = intOp.mapFlowable((Flowable) innerPublisher);
      }
```

still has 2 unchecked warnings, but IMO it's clearer why those warnings cannot be avoided: because the type system can't express a chain of IntermediateOperations mapping Object -> X -> Y -> ... -> R).